File: test/cases/record/test_record_with_function.c
#include <stdio.h>
typedef struct {
    int age;
    float score;
} person;
person p;

int maxage;


int getmax(int a, int b);

int getmax(int a, int b) {
    int _retval;
    if ((a > b)) {
        _retval = a;
    } else {
        _retval = b;
    }

    return _retval;
}


int main(void) {
    p.age = 25;
    p.score = 95.5;
    maxage = getmax(p.age, 30);
    printf("%d", maxage);

    return 0;
}

File: test/cases/record/test_record_with_function.pas
program test;
type
  person = record
    age: integer;
    score: real
  end;
var
  p: person;
  maxAge: integer;

function getMax(a, b: integer): integer;
begin
  if a > b then
    getMax := a
  else
    getMax := b
end;

begin
  p.age := 25;
  p.score := 95.5;
  maxAge := getMax(p.age, 30);
  write(maxAge)
end.
